### DISCUSSION TITLE
Easier way to debug with vscode breakpoint.

### DISCUSSION THREAD

**Discussion by quroom** [cookiecutter/cookiecutter-django#4998] [Ideas]
As long as we can't use we werkzeug debugger and debugpy, also there is no option disable werkzeug debugger with runserver_plus.
So I just turn runserver_plus to runserver.
And then the debugger works well.
umm.. I guess it's better to support choice which command we will use in only develop step.
How do you guys solve this issue?

**Comment by foarsitter** [comment]
> As long as we can't use we werkzeug debugger and debugpy

On windows (with WSL?) you may be correct but using Linux doesn't have these issues. A view years ago I had the same issues on my private laptop. I solved the issues by installing Ubuntu so that is something you may consider.

**Reply by quroom** [reply]
@foarsitter Exactaly.
Then what is best way better dev experience to integrate with vscode?
How do you debug with breakpoint?
Is it only to use linux without wsl? I mean directly install linux to my pc.
Or is there any another way to debug with breakpoint and at that time showing variables..?
It's poor developer experience in windows :(

**Reply by quroom** [reply]
And I have one more question. Is there no problem using both werkzeug debugger and debugpy with diffrent port in linux without wsl?
As you said, if it's ture, I will reall consider windows to linux for my dev pc.

**Reply by foarsitter** [reply]
You can try a dual boot first, at least, that is what I did. 

I'm using the pycharm debugger so I  have no experience with the others but if it runs on WSL it will be running on Linux (only better I suppose).

**Reply by quroom** [reply]
Thanks!

**Reply by quroom** [reply]
Reference. https://github.com/microsoft/debugpy/issues/1392
I can fix this problem.
